### PR TITLE
Replace CIDR-matching with dns magic for faster cdn detection

### PR DESCRIPTION
**Pull request by Klipsis**

### CODE REVIEW

File: src/worker-vless.js [hunk 1]
```diff
@@ -436,39 +466,55 @@ async function isCloudFlareIP(addressType, addressRemote) {
...
+	try {
+		const response = await fetch(`${doh}?name=${domain}.cdn.cloudflare.net&type=AAAA`, {
+			method: "GET",
+			headers: {
+				"Accept": "application/dns-json"
+			}
+		});
+		const data = await response.json();
+		const ans = data?.Answer;
+		//https://developers.cloudflare.com/1.1.1.1/encryption/dns-over-https/make-api-requests/dns-json/
```
Comment (zizifn) [inline src/worker-vless.js:493]: Do we need put  domainByoListCheck(domain, byoList) before doh fetch?

Comment (Klipsis) [reply src/worker-vless.js:493]: Sure thing. And tp improve accuracy we can replace the line 494 with: ```		return ans?.filter((record) => record.name === `${domain}.cdn.cloudflare.net` && record.type === 28).length > 1 || domainByoListCheck(domain, byoList); ```

Comment (zizifn) [reply src/worker-vless.js:493]: I mean, if we move `domainByoListCheck(domain, byoList)` before doh fetch, we don't need call doh for byo ip domain?

Comment (Klipsis) [reply src/worker-vless.js:493]: Both ways would work but actually byoCheck won't hit in the most of case.

**Comment by rikka0w0** [comment]
Does byo domain still work in early 2024?

**Pull request merged**